fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1420,808 @@ def draw_act(self, obj, layout, value):
+
+    @property
+    def bhv_object(self):
+        if not self.export_bhv:
+            return None
+        if self.export_all_selected:
+            return self.context_obj or bpy.context.active_object
+        else:
+            return self.col_object or self.gfx_object or self.context_obj or bpy.context.active_object
+
+    @property
+    def group_num(self):
+        if self.group_name == "common0":
+            return 1
+        else:
+            return int(self.group_name.removeprefix("group")) + 1
+
+    @property
+    def obj_name_col(self):
+        if self.export_all_selected:
+            return self.filter_name(self.col_object.name)
+        if not self.object_name and not self.col_object:
+            return ""
+        else:
+            return self.filter_name(self.object_name or self.col_object.name)
+
+    @property
+    def obj_name_gfx(self):
+        if self.export_all_selected:
+            return self.filter_name(self.gfx_object.name)
+        if not self.object_name and not self.gfx_object:
+            return ""
+        else:
+            return self.filter_name(self.object_name or self.gfx_object.name)
+
+    @property
+    def obj_name_bhv(self):
+        if self.export_all_selected:
+            return self.filter_name(self.bhv_object.name)
+        if not self.object_name and not self.bhv_object:
+            return ""
+        else:
+            return self.filter_name(self.object_name or self.bhv_object.name)
+
+    @property
+    def bhv_name(self):
+        return "bhv" + "".join([word.title() for word in toAlnum(self.obj_name_bhv).split("_")])
+
+    @property
+    def geo_name(self):
+        return f"{toAlnum(self.obj_name_gfx)}_geo"
+
+    @property
+    def collision_name(self):
+        return f"{toAlnum(self.obj_name_col)}_collision"
+
+    @property
+    def model_id_define(self):
+        return f"MODEL_{toAlnum(self.obj_name_gfx)}".upper()
+
+    @property
+    def export_level_name(self):
+        if self.level_name == "custom":
+            return self.custom_export_name
+        return self.level_name
+
+    # remove user prefixes/naming that I will be adding, such as _col, _geo etc.
+    def filter_name(self, name):
+        if self.use_name_filtering:
+            return sub("(_col)?(_geo)?(_bhv)?(lision)?", "", name)
+        else:
+            return name
+
+    def draw_generic_options(self, layout):
+        split = layout.split(factor=0.333)
+        if self.export_col:
+            box = split.box()
+            box.prop(self, "export_col")
+            box.prop(self, "include_children")
+            box.prop(self, "export_rooms")
+        else:
+            split.prop(self, "export_col")
+        if self.export_gfx:
+            box = split.box()
+            box.prop(self, "export_gfx")
+            box.prop(self, "export_script_loads")
+        else:
+            split.prop(self, "export_gfx")
+        if not self.export_all_selected:
+            split.prop(self, "export_bhv")
```

Comment:
Another ui suggestion, have you tried a row with align and then using toggle=1 for the export_ props? I tried it and it looks very clean imo, oot does it a lot and i love it (also using .column())
![image](https://github.com/Fast-64/fast64/assets/87947656/f8612059-b408-4184-a2cc-46a9d052380d)